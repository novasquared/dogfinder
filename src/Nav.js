import React from "react";
import { NavLink } from "react-router-dom";

/**
 * Creates nav links for dog list page and each dog
 * 
 * State: none
 * Props: dogs [{dog1}, ...]
 * 
 * App -> Nav
 */
function Nav({ dogs }) {
  return (
    <ul>
      <li><NavLink to="/DogList">Dogs</NavLink></li>
      {
        dogs.map(dog => {
          return <li key={dog.name}><NavLink to={`/dogs/${dog.src}`}>{dog.name}</NavLink></li>
        })
      }
    </ul>
  );
}
// end

export default Nav;