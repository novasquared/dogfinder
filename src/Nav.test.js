import React from 'react';
import Nav from './Nav';
import {render} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
const DOGS = [
  {
    "name": "Whiskey",
    "age": 5,
    "src": "whiskey",
    "facts": [
      "Whiskey loves eating popcorn.",
      "Whiskey is a terrible guard dog.",
      "Whiskey wants to cuddle with you!"
    ]
  },
  {
    "name": "Duke",
    "age": 3,
    "src": "duke",
    "facts": [
      "Duke believes that ball is life.",
      "Duke likes snow.",
      "Duke enjoys pawing other dogs."
    ]
  },
  {
    "name": "Perry",
    "age": 4,
    "src": "perry",
    "facts": [
      "Perry loves all humans.",
      "Perry demolishes all snacks.",
      "Perry hates the rain."
    ]
  }
]

// basic tests
it('renders without crashing', function() {
  render(
    <MemoryRouter>
      <Nav dogs={DOGS}/>
    </MemoryRouter>
  );
});

it('matches snapshot', function() {
  const container = render(
    <MemoryRouter>
      <Nav dogs={DOGS}/>
    </MemoryRouter>
  );
  expect(container.asFragment()).toMatchSnapshot();
});
// end basic tests

// full render
it('mounts without crashing', function() {
  const { getByText } = render(
    <MemoryRouter>
      <Nav dogs={DOGS}/>
    </MemoryRouter>
  );

  //checks dog list
  const DogListLink = getByText(/Dogs/i);
  expect(DogListLink).toBeInTheDocument();

  //checks single dog
  const Dog = getByText(/whiskey/i);
  expect(Dog).toBeInTheDocument();
});
