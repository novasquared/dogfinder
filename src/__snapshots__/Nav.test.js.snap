// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/DogList"
      >
        Dogs
      </a>
    </li>
    <li>
      <a
        href="/dogs/whiskey"
      >
        Whiskey
      </a>
    </li>
    <li>
      <a
        href="/dogs/duke"
      >
        Duke
      </a>
    </li>
    <li>
      <a
        href="/dogs/perry"
      >
        Perry
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
